build.rs

```diff
@@ -63,8 +58,56 @@ fn configured_by_vcpkg() -> bool {
     false
 }
 
+fn static_or_not() {
+    use std::ascii::AsciiExt;
+
+    // On Windows-MSVC, always link dynamically
+    if cfg!(all(windows, target_env="msvc")) {
+        println!("cargo:rustc-link-lib=dylib=libpq");
+        return;
+    }
+
+    // Link unconditionally statically
+    if env::var_os("PQ_LIB_STATIC").is_some() {
+        println!("cargo:rustc-link-lib=static=pq");
+        return;
+    }
+
+    // Examine the per-target env vars
+    if let Ok(target) = env::var("TARGET") {
+        let pg_config_for_target = format!("PQ_LIB_STATIC_{}", target.to_ascii_uppercase().replace("-", "_"));
+        if env::var_os(&pg_config_for_target).is_some() {
+            println!("cargo:rustc-link-lib=static=pq");
+            return;
+        }
+    }
+    
+    // Otherwise, don't specify
+    println!("cargo:rustc-link-lib=pq");
+}
+
+fn pg_config_path() -> PathBuf {
+    use std::ascii::AsciiExt;
+
+    if let Ok(target) = env::var("TARGET") {
+        let pg_config_for_target = &format!("PG_CONFIG_{}", target.to_ascii_uppercase().replace("-", "_"));
```

Comment:
This also requires something like:

```rust
println!("cargo:rerun-if-env-changed={}", pg_config_for_target);
```

It might be best to wrap `env::var_os` in a wrapper that always generates this `println!`, and then refactor all of `build.rs` to use it.